RIOT-OS/RIOT
# pyterm: add native support

### Contribution description

I always found it annoying that native is lacking the convenience features of `pyterm` (e.g., history, concatenating of commands, logging etc). Of course, one could pipe stdin/stdout into something like socat and use pyterm's feature to connect to a TCP server (e.g., `socat EXEC:${NATIVEBIN},end-close,stderr,pty TCP-LISTEN:4711,reuseaddr,fork & pyterm -ts 4711`) but that's somewhat tedious and not well-aligned with the typical usage.

Hence, this PR will modify the `term` make target for native so that pyterm is called instead of calling the native binary directly. In turn, pyterm now has a new option which will spawn RIOT native processes as a subprocess (i.e., child process) and communicate to it via pipes.

### Testing procedure

Call `BOARD=native make term` for arbitrary RIOT applications.

Caution: this is not well tested so far - expect the unexpected.
